### DISCUSSION TITLE
OpenThread configuration options and their descriptions

### DISCUSSION THREAD

**Discussion by dsyx** [openthread/openthread#9572] [Q&A] [answered]
Hello everyone,

In [OpenThread - BUILD](https://openthread.io/guides/build), the **Configuration** section mentions CMake build options that are used to enable various features of OpenThread. However, the documentation [openthread/examples/README.md](https://github.com/openthread/openthread/blob/main/examples/README.md) is currently deleted.

I would like to get all the configuration options, their detailed descriptions and when it is recommended to enable them.

Thanks

**Comment by superwhd** [comment]
You may refer to https://github.com/openthread/openthread/blob/main/etc/cmake/options.cmake for the available CMake options.

https://github.com/openthread/openthread/blob/main/script/cmake-build#L73 contains some common build options used in github actions.

**Reply by dsyx** [reply]
Grateful,

Is there documentation explaining these options in detail? For example, which options are recommended for FTD devices that comply with the Thread 1.3 standard, and which options are recommended for devices that need to use COAP.

Thanks

**Reply by jwhui** [reply, marked as answer]
CMake options generally map to build options defined in [src/core/config](https://github.com/openthread/openthread/tree/main/src/core/config), which has documentation.

Default configuration for Thread 1.3 should provide a Thread 1.3 compliant implementation.

**Reply by dsyx** [reply]
Thanks
